code/modules/food_and_drinks/kitchen_machinery/icecream_vat.dm

```diff
@@ -18,6 +19,7 @@
 	var/list/product_types = list()
 	var/dispense_flavour = ICECREAM_VANILLA
 	var/flavour_name = "vanilla"
+	var/obj/item/reagent_containers/beaker = null
```

Comment:
I believe you are allowing any reagent container to work not just beakers. Keep in mind a lot of things are classified as reagent containers.

Reply:
This means you can make pizza flavoured icecream, you tell me that isn't a hilarious notion.

Reply:
I don't think the machine will accept any non-open containers such as food, pills or patches.

Reply:
Ah I see now

Reply:
the last `/` is the var name so the `beaker` var is typed as `/obj/item/reagent_containers`